vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceTextBlockManager.java

```diff
@@ -0,0 +1,90 @@
+import edu.asu.diging.vspace.core.model.display.ISpaceTextBlockDisplay;
+import edu.asu.diging.vspace.core.model.display.impl.ModuleLinkDisplay;
+import edu.asu.diging.vspace.core.model.display.impl.SpaceTextBlockDisplay;
+import edu.asu.diging.vspace.core.model.impl.ModuleLink;
+import edu.asu.diging.vspace.core.model.impl.SpaceTextBlock;
+import edu.asu.diging.vspace.core.services.ISpaceManager;
+import edu.asu.diging.vspace.core.services.ISpaceTextBlockManager;
+
+@Transactional
+@Service
+public class SpaceTextBlockManager implements ISpaceTextBlockManager{
+    @Autowired
+    private ISpaceManager spaceManager;
+
+    @Autowired
+    private ISpaceTextBlockFactory spaceTextBlockFactory;
+
+    @Autowired
+    private ISpaceTextBlockDisplayFactory spaceTextBlockDisplayFactory;
+
+    @Autowired
+    private SpaceTextBlockRepository spaceTextBlockRepo;
+
+    @Autowired
+    private SpaceTextBlockDisplayRepository spaceTextBlockDisplayRepo;
+
+    @Override
+    public ISpaceTextBlockDisplay createTextBlock(String id, float positionX, float positionY, String text,
+            float height, float width)
+                    throws SpaceDoesNotExistException, ImageCouldNotBeStoredException, SpaceDoesNotExistException {
+        ISpace source = spaceManager.getSpace(id);
+        ISpaceTextBlock textBlock=spaceTextBlockFactory.createSpaceTextBlock(text, source);
+        ISpaceTextBlockDisplay spaceTextBlockDisplay = spaceTextBlockDisplayFactory.createSpaceTextBlockDisplay(textBlock,positionX,positionY,height, width);
+        textBlock=spaceTextBlockRepo.save((SpaceTextBlock) textBlock);
+        return spaceTextBlockDisplayRepo.save((SpaceTextBlockDisplay)spaceTextBlockDisplay);
+    }
+
+    @Override
+    public List<ISpaceTextBlockDisplay> getSpaceTextBlockDisplays(String spaceId) {
+        return spaceTextBlockDisplayRepo.findSpaceTextBlockDisplaysForSpace(spaceId);
+    }
+    
+    @Override
+    public void deleteTextBlock(String blockId) {
+        Optional<SpaceTextBlock> spaceTextBlock = spaceTextBlockRepo.findById(blockId);
+        if(spaceTextBlock.isPresent()) {
+            spaceTextBlockDisplayRepo.deleteBySpaceTextBlock(spaceTextBlock.get());
+            spaceTextBlockRepo.delete(spaceTextBlock.get());
+        }
+    }
+    @Override
+    public ISpaceTextBlockDisplay updateTextBlock(String id, Float positionX, Float positionY, String textBlockIdValueEdit,
+            String textBlockDisplayId, String text, Float height, Float width) {
+        
+        Optional<SpaceTextBlock> spaceTextBlock = spaceTextBlockRepo.findById(textBlockIdValueEdit);
+        if(spaceTextBlock.isPresent()) {
+            ISpaceTextBlock spaceTextBlockEdit = spaceTextBlock.get();
+            spaceTextBlockEdit.setText(text);
+            Optional<SpaceTextBlockDisplay> spaceTextBlockDisplay = spaceTextBlockDisplayRepo.findById(textBlockDisplayId);
+            if(spaceTextBlockDisplay.isPresent()) {
```

Comment:
It would be a lot either to understand if you first get the block and the display, return if one of them doesn't exist; if both exist, do the update.